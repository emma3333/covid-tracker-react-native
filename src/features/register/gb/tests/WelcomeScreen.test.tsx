import renderer from 'react-test-renderer'
import React from 'react'
import { WelcomeScreen } from '../WelcomeScreen'
import { render, fireEvent } from 'react-native-testing-library'
import { Linking } from 'react-native'

describe('Welcome Screen', () => {
    it('should match snapshot', () => {
        const tree = renderer.create(
                <WelcomeScreen />
            ).toJSON()
        expect(tree).toMatchSnapshot()    
    })

    test('Visit NHS website link fires onPress event', () => {
        const visitNhsText = 'visit the NHS website'
        const onPressMock = jest.fn(() =>
          Linking.openURL('https://www.nhs.uk/conditions/coronavirus-covid-19/')
        )
        const { getByTestId } = render(
            <WelcomeScreen testID='nhsLink' onPress={onPressMock} />
        )
        fireEvent(getByTestId('nhsLink'), 'onPress', visitNhsText)
        expect(onPressMock).toHaveBeenCalled()
        expect(onPressMock).toHaveBeenCalledWith(visitNhsText)
    })

    test('Create account fires onPress event', () => {
        const createAccountText = 'Create account'
        const onPressMock = jest.fn()
        const { getByTestId } = render(<WelcomeScreen testID='createAccount' onPress={onPressMock} />)      
        fireEvent(getByTestId('createAccount'), 'onPress', createAccountText)
        expect(onPressMock).toHaveBeenCalled()
        expect(onPressMock).toHaveBeenCalledWith(createAccountText)
    })

    test('For more info fires onPress event', () => {
        const moreInfoText = 'For more info visit our website'
        const onPressMock = jest.fn()
        const { getByTestId } = render(<WelcomeScreen testID='moreInfo' onPress={onPressMock} />)
        fireEvent(getByTestId('moreInfo'), 'onPress', moreInfoText)
        expect(onPressMock).toHaveBeenCalled()
        expect(onPressMock).toHaveBeenCalledWith(moreInfoText)
    })

})
