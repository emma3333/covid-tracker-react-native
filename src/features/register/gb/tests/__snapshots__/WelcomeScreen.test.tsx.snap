// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Welcome Screen should match snapshot 1`] = `
<RCTScrollView
  contentContainerStyle={
    Object {
      "flexGrow": 1,
      "justifyContent": "space-between",
    }
  }
>
  <View>
    <View
      style={
        Object {
          "backgroundColor": "#024364",
          "flex": 1,
        }
      }
    >
      <View
        style={
          Object {
            "paddingHorizontal": 24,
            "paddingVertical": 24,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "space-between",
            }
          }
        >
          <Image
            resizeMode="contain"
            source={1}
            style={
              Object {
                "height": 60,
                "width": 60,
              }
            }
          />
          <Text
            style={
              Object {
                "color": "#FFFFFF",
                "fontSize": 14,
                "paddingHorizontal": 5,
              }
            }
          >
            COVID Symptom Tracker
          </Text>
          <View
            accessible={true}
            focusable={true}
            isTVSelectable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <Image
              source={1}
              style={
                Object {
                  "height": 32,
                  "width": 32,
                }
              }
            />
          </View>
          <View
            style={
              Object {
                "alignSelf": "center",
                "fontWeight": "300",
              }
            }
          >
            <Text
              onPress={[Function]}
              style={
                Array [
                  Object {
                    "color": "#A10056",
                    "fontSize": 16,
                    "lineHeight": 24,
                  },
                  Object {
                    "color": "#FFFFFF",
                    "fontWeight": "700",
                  },
                ]
              }
            >
              Sign in
            </Text>
          </View>
        </View>
        <View>
          <Text
            style={
              Array [
                Object {
                  "color": "#565A5C",
                  "fontSize": 16,
                  "lineHeight": 24,
                },
                Object {
                  "color": "#FFFFFF",
                  "fontSize": 24,
                  "lineHeight": 38,
                  "marginTop": 15,
                  "paddingVertical": 24,
                  "textAlign": "center",
                },
              ]
            }
          >
            Take 1 minute each day and help fight the outbreak.
          </Text>
          <Text
            style={
              Array [
                Object {
                  "color": "#565A5C",
                  "fontSize": 16,
                  "lineHeight": 24,
                },
                Object {
                  "color": "#FFFFFF",
                  "fontSize": 14,
                  "fontWeight": "300",
                  "lineHeight": 20,
                  "textAlign": "center",
                },
              ]
            }
          >
            

            This app allows you to help others, but does 
            

             not give health advice. If you need health 
            

             advice please
             
            <Text
              onPress={[Function]}
              style={
                Array [
                  Object {
                    "color": "#A10056",
                    "fontSize": 16,
                    "lineHeight": 24,
                  },
                  Array [
                    Object {
                      "color": "#FFFFFF",
                      "fontSize": 14,
                      "fontWeight": "300",
                      "lineHeight": 20,
                      "textAlign": "center",
                    },
                    Object {
                      "textDecorationLine": "underline",
                    },
                  ],
                ]
              }
            >
              visit the NHS website
            </Text>
            .
            

            

          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "alignContent": "center",
            "backgroundColor": "#FFFFFF",
            "borderTopLeftRadius": 20,
            "borderTopRightRadius": 20,
            "flex": 1,
            "justifyContent": "space-between",
            "padding": 32,
          }
        }
      >
        <Image
          resizeMode="contain"
          source={1}
          style={
            Object {
              "alignSelf": "center",
              "height": 120,
              "marginBottom": 20,
              "marginTop": 30,
              "resizeMode": "center",
            }
          }
        />
        <View
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "alignSelf": "stretch",
              "backgroundColor": "#024364",
              "borderBottomWidth": null,
              "borderColor": "#007aff",
              "borderLeftWidth": null,
              "borderRadius": 100,
              "borderRightWidth": null,
              "borderTopWidth": null,
              "borderWidth": undefined,
              "elevation": 0,
              "flexDirection": "row",
              "height": 56,
              "justifyContent": "center",
              "opacity": 1,
              "paddingBottom": 6,
              "paddingTop": 6,
              "shadowColor": undefined,
              "shadowOffset": undefined,
              "shadowOpacity": undefined,
              "shadowRadius": undefined,
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#565A5C",
                  "fontSize": 16,
                  "lineHeight": 24,
                },
                Object {
                  "color": "#FFFFFF",
                },
              ]
            }
          >
            Create account
          </Text>
        </View>
        <Text
          onPress={[Function]}
          style={
            Array [
              Object {
                "color": "#A10056",
                "fontSize": 16,
                "lineHeight": 24,
              },
              Object {
                "fontSize": 14,
                "lineHeight": 20,
                "marginTop": 10,
                "textAlign": "center",
              },
            ]
          }
        >
          For more info 
          <Text
            style={
              Array [
                Object {
                  "color": "#565A5C",
                  "fontSize": 16,
                  "lineHeight": 24,
                },
                Object {
                  "color": "#A10056",
                  "fontSize": 14,
                  "lineHeight": 20,
                },
              ]
            }
          >
            visit our website
          </Text>
        </Text>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
